lib/Frontend/OutputFileMap.cpp

```diff
@@ -105,7 +105,7 @@ void OutputFileMap::dump(llvm::raw_ostream &os, bool Sort) const {
 static void writeQuotedEscaped(llvm::raw_ostream &os,
                                const StringRef fileName) {
   os << "\"";
-  os.write_escaped(fileName);
+  os.write_escaped(fileName, true);
```

Comment:
It would be nice™ to use a pseudo-keyword-argument comment here: `write_escaped(fileName, /*UseHexEscapes=*/true)`.

(Obviously it would be even better if the original API had an enum or something, but this would still be an improvement.)

If you do make this change, I don't think you'd need to run the source compat suite tests again; just watch to see that the previous build completes successfully. Or save it for a follow-up, since this is getting the bots blue again.

Reply:
Good point! I'll make this change. Have also enhanced a test.